api/src/main/java/org/pmiops/workbench/interceptors/TracingInterceptor.java

```diff
@@ -45,28 +51,57 @@ public TracingInterceptor(Provider<WorkbenchConfig> workbenchConfigProvider) {
   @Override
   public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
       throws Exception {
+    // OPTIONS methods requests don't need traces.
+    if (request.getMethod().equals(HttpMethods.OPTIONS)) {
+      return true;
+    }
+    // All of our Swagger-generated handlers extend the HandlerMethod class, but we'll check this
+    // before casting to avoid runtime exceptions.
+    if (!(handler instanceof HandlerMethod)) {
+      return true;
+    }
 
+    // Create a new scoped span, which will set the span context for any other traces created
+    // from within this request-handling thread.
+    //
+    // This span will end up in Stackdriver with the name format like "Recv.[methodName]" where
+    // methodName is the method name defined in our Swagger YAML.
+    // Example: Recv.syncBillingProjectStatus
+    HandlerMethod handlerMethod = (HandlerMethod) handler;
     SpanBuilder requestSpanBuilder =
-        tracer.spanBuilder(
-            String.format(
-                "%s/%s%s",
-                workbenchConfigProvider.get().server.shortName,
-                request.getMethod(),
-                request.getRequestURI()));
-
+        tracer.spanBuilder(handlerMethod.getMethod().getName()).setSpanKind(Span.Kind.SERVER);
     if (workbenchConfigProvider.get().server.traceAllRequests) {
       requestSpanBuilder.setSampler(Samplers.alwaysSample());
     }
+    Scope requestScopedSpan = requestSpanBuilder.startScopedSpan();
 
-    Scope requestSpan = requestSpanBuilder.startScopedSpan();
-    request.setAttribute(TRACE_ATTRIBUTE_KEY, requestSpan);
+    // Log some additional key-value attributes, which will be visible in the details pane on
+    // Stackdriver.
+    final Span currentSpan = tracer.getCurrentSpan();
+    currentSpan.putAttribute(
```

Comment:
I think the env is actually implicit in the stackdriver project we see it in. Although this could help with screenshots. (I have no objections with this being here, just noting)

Reply:
Ack – good point. I'll leave it here though, because for local testing, traces go to Test project but we want to see the "local" indication somewhere at least.